cogs/channel_checker.py

```diff
@@ -0,0 +1,82 @@
+import asyncio
+import difflib
+
+import discord
+from discord.ext.commands import Bot, Cog
+
+from config import CONFIG
+
+
+def locate(channel_name, channel_list):
+    """Find the channel from id, and create location string ("between #abc and #def")"""
+    for i, c in enumerate(channel_list):
+        if c.name == channel_name:
+            if i == 0:
+                msg = f"before {channel_list[1].mention}"
+            if i == len(channel_list) - 1:
+                msg = f"after {channel_list[len(channel_list)-2].mention}"
+            else:
+                msg = (
+                    f"between {channel_list[i-1].mention} "
+                    f"and {channel_list[i+1].mention}"
+                )
+            return c, msg
+    return None, "somewhere"
+
+
+def channel_sort(channels):
+    """Sort text channels into actual order"""
+    channels = [c for c in channels if isinstance(c, discord.TextChannel)]
+    return sorted(channels, key=lambda c: c.position)
+
+
+async def channel_check(bot):
+    """Periodically checks for channels that have been reordered"""
+    await bot.wait_until_ready()
+
+    guild = bot.get_guild(CONFIG.UWCS_DISCORD_ID)
+    channel = bot.get_channel(CONFIG.UWCS_EXEC_SPAM_CHANNEL_ID)
+
+    previous = channel_sort(guild.channels)
+    while not bot.is_closed():
+        await asyncio.sleep(CONFIG.CHANNEL_CHECK_INTERVAL)
+
+        # Get channel ids for diff
+        current = channel_sort(guild.channels)
+        curr_channels = [c.name for c in current]
+        prev_channels = [c.name for c in previous]
+
+        if curr_channels == prev_channels:
+            continue
+
+        # Find and filter changes
+        changes = list(difflib.Differ().compare(prev_channels, curr_channels))
+
+        added = [c.strip("+ ") for c in changes if c[0] == "+"]
+        removed = [c.strip("- ") for c in changes if c[0] == "-"]
+
+        moved = [c for c in added if c in removed]  # Moved if added and removed
+
+        # Construct message
+        if moved:
+            msg = "**⚠️ Channel Moved:**"
+
+            if moved:
+                for channel_name in moved:
+                    c, prev_pos_str = locate(channel_name, previous)
+                    c, curr_pos_str = locate(channel_name, current)
+                    msg += f"\n\t{c.mention} has been moved from {prev_pos_str} to {curr_pos_str}"
+
+            await channel.send(msg)
+
+        previous = current
+
+
+class ChannelChecker(Cog):
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        self.bot.loop.create_task(channel_check(self.bot))
```

Comment:
Another approach would be to use the [relevant event](https://discordpy.readthedocs.io/en/stable/api.html#discord.on_guild_channel_update) and then retrieve the neighbours, accounting for whether the channel moved up/down.

Reply:
Turns out proper synchronization is a few separate kettles of fish, primarily: 
1) Each channel update only updates the position of that channel, so can lead to repeated positions
2) These events arrive in no particular order, so no easy way to tell what the main move is (best way is probably to take the sequence of updates and find which channel is moved over the largest range, not foolproof though)
3) And in general, also need to consider this https://github.com/Rapptz/discord.py/issues/2392
Seems like the easiest solution is to wait some time (idk 5s) until after the first order update, then properly process them at the same time. Still need to account for (3) though

Reply:
As an aside, we have several looping tasks throughout Apollo. A future PR may entail factoring out this common behaviour into a class that can handle them all.